### DISCUSSION TITLE
How to publish GrapesJs layout on webpage

### DISCUSSION THREAD

**Discussion by codingmachine16** [GrapesJS/grapesjs#3666] [General]
Hi @artf 

Our GrapesJs editor is running on 
http://localhost:5500/editor

![image](https://user-images.githubusercontent.com/87995581/128018584-1ba2bc43-aeca-4e14-90dd-7bb986816072.png)

We want to publish it on front end on 
http://localhost:5500/

But on front end , the panels are showing and edit mode is on. How to disable edit mode and hide panels

![image](https://user-images.githubusercontent.com/87995581/128018838-f0128973-0a49-4092-b91f-1975577c1f1a.png)

**Comment by artf** [comment]
The final page has nothing to do with the editor. For example, you store HTML/CSS somewhere in files/DB from `/editor` page, then in your `/` page you just need to get the final HTML/CSS from when you have stored them and print them as output, you don't need to include GrapesJS there...

**Reply by codingmachine16** [reply]
It is in this format 

{"mycustom-html":"<header class=\"bg-dark\"><div class=\"container-fluid\"><div></div><div class=\"row\" id=\"iaul\"><hr class=\"d-none\"/></div></div><div></div></header>","mycustom-components":"[{\"tagName\":\"header\",\"classes\":[\"bg-dark\"],\"components\":[{\"classes\":[\"container-fluid\"],\"components\":[{},{\"classes\":[{\"name\":\"row\",\"private\":1}],\"attributes\":{\"id\":\"iaul\"},\"components\":[{\"tagName\":\"hr\",\"void\":true,\"classes\":[\"d-none\"]}]}]},{}]}]","mycustom-assets":"[]","mycustom-css":"* { box-sizing: border-box; } body {margin: 0;}#iaul{min-height:75px;}","mycustom-styles":"[{\"selectors\":[\"#iaul\"],\"style\":{\"min-height\":\"75px\"}}]"}

How can we get this and print? I am using Reactjs. We are stuck here though we have completed a website builder using grapesJs.

**Reply by artf** [reply]
Sorry mate, but I think questions like this you have to ask on sites like StackOverflow, this is strictly related to your app and has nothing to do with GrapesJS...

**Reply by codingmachine16** [reply]
@artf 

I have found the solution. Thanks alot guidance.

**Reply by IreAy0** [reply]
how did you achieve this, please?

**Reply by codingmachine16** [reply]
@IreAy0 

// got home page data from DB using node Js
    getpage("home") 
      .then((data) => {
        setTimeout(() => {
          setPageContent(data); // stored data in React State
        }, 500);
      })
      .catch((err) => console.log(err));
  }

  useEffect(() => {
    getPage();  // getting data from DB
  }, []);

// Now filled stored HTML from DB to webpage <div id="root"> </div>
  document.getElementById("root").innerHTML = pageContent["mycustom-html"];
